addons/l10n_no_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _
+from odoo.addons.account_edi_ubl_bis3.models.account_edi_format import COUNTRY_EAS
+
+import base64
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    ####################################################
+    # Import
+    ####################################################
+
+    def _is_ubl(self, filename, tree):
+        """ OVERRIDE so that the generic ubl parser does not parse BIS3 any longer.
+        """
+        is_ubl = super()._is_ubl(filename, tree)
+        return is_ubl and not self._is_ehf_3(filename, tree)
```

Comment:
this is strange... why not
```
if self.code == '<truc_no_edi>':
    return self._is_ehf_3(filename, tree)
else:
    return super()._is_ubl(filename, tree)
```

Reply:
`self.code` has nothing to do here, it's already checked in `_create_invoice_from_xml_tree` and `_update_invoice_from_xml_tree` that calls this function. 

ubl is more general than ehf, therefore if the parser of ubl goes first, it will parse the file even though it should be parsed by the ehf decoder, to yield a better result. This restricts our detection of generic ubl so it ignores profiles that should be decoded by a dedicated decoder.